pkgs/applications/misc/pdfstudio/default.nix

```diff
@@ -0,0 +1,75 @@
+{ stdenv,
+  lib,
+  makeWrapper,
+  fetchurl,
+  dpkg,
+  autoPatchelfHook,
+  gst_all_1,
+  sane-backends,
+  xorg,
+  gnome2,
+  alsa-lib,
+  libgccjit,
+  }:
+
+let
+  year = "2021";
+  rel = "1";
+  subrel = "1";
+in
+stdenv.mkDerivation rec {
+  pname = "pdfstudio";
+  version = "${year}.${rel}.${subrel}";
+  autoPatchelfIgnoreMissingDeps=true;
+
+  # fetching a .deb because there's no easy way to package this Electron app
+  src = fetchurl {
+    url = "https://download.qoppa.com/${pname}/v${year}/PDFStudio_v${year}_${rel}_${subrel}_linux64.deb";
+    sha256 = "089jfpbsxwjhx245g8svlmg213kny3z5nl6ra1flishnrsfjjcxb";
+  };
+
+  nativeBuildInputs = [
+    gst_all_1.gst-libav
+    sane-backends
+    xorg.libXxf86vm
+    xorg.libXtst
+    gnome2.libgtkhtml
+    alsa-lib
+    libgccjit
+    autoPatchelfHook
+    makeWrapper
+    dpkg
+  ];
+
+  unpackPhase = "dpkg-deb -x $src .";
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share
+    mkdir -p $out/share/applications
+    mkdir -p $out/share/pixmaps
+    cp -r opt/${pname}${year} $out/share/
+    cp ${./pdfstudio.desktop} $out/share/applications/${pname}.desktop
+    cp ${./pdfstudio.png}  $out/share/pixmaps/
+    makeWrapper $out/share/${pname}${year}/${pname}${year} $out/bin/${pname}
+  '';
+
+  # We need to run pdfstudio once. This will unpack some jre.pack files in jre/lib.
+  # We want to do this after autoPatchelfHook, so we'll do this in preInstallCheck.
+  # Alternatively, we can patchelf manually, using this (e.g., in the fixup phase):
+  # find $out/share/${pname}${year}/jre/bin -type f -executable -exec patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) {} \;
+  doInstallCheck = true;
+  preInstallCheck = ''
```

Comment:
This should probably done in `postFixup` (if it works)?

Reply:
This does not work, because autoPatchelfHook needs to run before, and (suprisingly) it doesn't before postFixup.

Reply:
I just looked at the sources of `autoPatchelfHook` it hooks itself into the `postFixup` phase and that seem occur just before the definition of the `postFixup` code in the derivation. However, in your comment you mention that you only need replace to replace the interpreter/dynamic linker. I think it would be cleaner here to use manual `patchelf` instread e.g. in the `preFixup` than using `autoPatchelf` here, since `autoPatchelfIgnoreMissingDeps` is to true here.